zebra-chain/src/transaction/serialize.rs

```diff
@@ -114,7 +114,7 @@ fn coinbase_height_len(height: BlockHeight) -> usize {
         3
     } else if let _h @ 65536..=16_777_215 = height.0 {
         4
-    } else if let _h @ 16_777_216..=499_999_999 = height.0 {
+    } else if let _h @ 16_777_216..=BlockHeight::MAX_AS_U32 = height.0 {
```

Comment:
isn't this a different condition? should it be `..BlockHeight::MAX_AS_U32` instead?